JAICore/jaicore-ml/src/test/java/ai/libs/jaicore/ml/scikitwrapper/ScikitLearnWrapperTest.java

```diff
@@ -1,192 +1,125 @@
-		ScikitLearnWrapper<ISingleLabelClassification, ISingleLabelClassificationPredictionBatch> slw = new ScikitLearnWrapper<>(constructInstruction, ScikitLearnWrapper.getImportString(imports), EScikitLearnProblemType.CLASSIFICATION);
-		ILabeledDataset<ILabeledInstance> datasetTest = this.loadARFF(CLASSIFICATION_ARFF);
-		slw.setModelPath(new File(CLASSIFIER_DUMP).getAbsoluteFile());
-		ISingleLabelClassificationPredictionBatch result = slw.predict(datasetTest);
-		assertEquals("Unequal length of predictions and number of test ILabeledDataset<ILabeledInstance>", result.getNumPredictions(), datasetTest.size());
-	}
-
-	@Test
-	@LongTest
-	public void getRawOutput() throws Exception {
-		ScikitLearnWrapper<IRegressionPrediction, IRegressionResultBatch> slw = new ScikitLearnWrapper<>("MLPRegressor()", "from sklearn.neural_network import MLPRegressor", EScikitLearnProblemType.REGRESSION);
-		ILabeledDataset<ILabeledInstance> datasetTrain = this.loadARFF(BAYESNET_TRAIN_ARFF);
-		ILabeledDataset<ILabeledInstance> datasetTest = this.loadARFF(BAYESNET_TRAIN_ARFF);
-		int s = datasetTrain.getNumAttributes();
-		slw.setTargets(s - 1, s - 2, s - 3);
-		slw.fit(datasetTrain);
-		slw.predict(datasetTest);
-		assertNotNull(MSG_MODELPATH_NOT_NULL, slw.getModelPath());
-		assertTrue(slw.getModelPath().exists());
-	}
-
-	@Test
-	@LongTest
-	public void invalidConstructorNoConstructionCall() throws IOException {
-		assertThrows(AssertionError.class, () -> {
-			new ScikitLearnWrapper<SingleLabelClassification, SingleLabelClassificationPredictionBatch>(null, "", EScikitLearnProblemType.CLASSIFICATION);
-		});
-	}
-
-	@Test
-	@LongTest
-	public void invalidConstructorEmptyConstructionCall() throws IOException {
-		assertThrows(AssertionError.class, () -> {
-			new ScikitLearnWrapper<SingleLabelClassification, SingleLabelClassificationPredictionBatch>("", "", EScikitLearnProblemType.CLASSIFICATION);
-		});
-	}
-
-	private ILabeledDataset<ILabeledInstance> loadARFF(final String arffPath) throws IOException, DatasetDeserializationFailedException, InterruptedException {
-		return ArffDatasetAdapter.readDataset(new File(arffPath));
-	}
-
-}
+package ai.libs.jaicore.ml.scikitwrapper;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.commons.io.FileUtils;
+import org.api4.java.ai.ml.core.dataset.supervised.ILabeledDataset;
+import org.api4.java.ai.ml.core.dataset.supervised.ILabeledInstance;
+import org.api4.java.ai.ml.core.evaluation.IPrediction;
+import org.api4.java.ai.ml.core.evaluation.IPredictionBatch;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.jupiter.api.Test;
+
+import ai.libs.jaicore.ml.core.dataset.serialization.ArffDatasetAdapter;
+
+public class ScikitLearnWrapperTest {
```

Comment:
should be split up into one test per setting. Maybe use a parameterized test here to avoid code duplication?